components/bl-audio-recorder-component/src/index.js

```diff
@@ -0,0 +1,153 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import { captureMediaDevices, download } from './helpers';
+
+const { cn, normalizeDimensionValue } = BackendlessUI.CSSUtils;
+
+export default function AudioRecorder({ component, eventHandlers, elRef }) {
+  const { player, controls, noise, fileName, fileType, width, display, style, classList } = component;
+  const { onStart, onStop, onDownload, onState } = eventHandlers;
+  const { paused, recording, inactive } = STREAM_STATES;
+
+  const audioRef = useRef();
+  const recorderRef = useRef();
+
+  const [recordedBlob, setRecordedBlob] = useState();
+  const [state, setState] = useState();
+
+  const styles = useMemo(() => ({
+    width: normalizeDimensionValue(width),
+    ...style,
+  }), [style]);
+
+  useEffect(() => {
+    if (recordedBlob && audioRef.current) {
+      audioRef.current.src = URL.createObjectURL(recordedBlob);
+    }
+  }, [recordedBlob]);
+
+  useEffect(() => {
+    if (state) {
+      onState({ state });
+    }
+  }, [state]);
+
+  Object.assign(component, {
+    start       : () => startRecording(),
+    stop        : () => stopRecording(),
+    downloadFile: () => downloadRecordedFile(),
+    getBlob     : () => recordedBlob,
+    pause       : () => toggleRecord(),
+  });
+
+  const startRecording = useCallback(async () => {
+    const audioConstrains = { audio: { echoCancelation: true, noiseSuppression: noise } };
+    const stream = await captureMediaDevices(audioConstrains, audioRef);
+
+    if (stream) {
+      recorderRef.current = new MediaRecorder(stream);
+
+      const chunks = [];
+
+      recorderRef.current.ondataavailable = event => {
+        if (event.data.size > 0) {
+          chunks.push(event.data);
+        }
+      };
+
+      recorderRef.current.onstart = () => {
+        setState(recorderRef.current.state);
+      };
+
+      recorderRef.current.onstop = () => {
+        const recordedBlob = new Blob(chunks, { type: TYPES[fileType] });
+
+        setState(recorderRef.current.state);
+        setRecordedBlob(recordedBlob);
+        onStop();
+
+        chunks.length = 0;
+      };
+
+      recorderRef.current.onpause = () => {
+        setState(recorderRef.current.state);
+      };
+
+      recorderRef.current.onresume = () => {
+        setState(recorderRef.current.state);
+      };
+
+      recorderRef.current.start();
+      onStart();
+    }
+  }, []);
```

Comment:
Are you sure we don't need there any deps? Noize/fileType?

Reply:
`noise` and `fileType` don't have data binding or logic, so they will not be changed.